clash-cores/src/Clash/Cores/Xilinx/Xpm/Cdc/ArraySingle/Internal.hs

```diff
@@ -0,0 +1,179 @@
+{-|
+  Copyright   :  (C) 2023, Google LLC
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE ViewPatterns #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Clash.Cores.Xilinx.Xpm.Cdc.ArraySingle.Internal where
+
+import Prelude
+import Clash.Explicit.Prelude
+  (type (<=), SNat, BitVector, Clock, KnownNat, KnownDomain, errorX, unsafeSynchronizer)
+
+import Clash.Annotations.Primitive (Primitive(..), HDL(..), hasBlackBox)
+import Clash.Backend (Backend)
+import Clash.Netlist.Types (TemplateFunction(..), BlackBoxContext)
+import Clash.Promoted.Nat (snatToNum)
+import Clash.Signal.Internal (Signal((:-)))
+
+import Control.Monad.State (State)
+import Data.List.Infinite (Infinite(..), (...))
+import Data.String.Interpolate (__i)
+import Data.Text (Text)
+import Data.Text.Prettyprint.Doc.Extra (Doc)
+import GHC.Stack (HasCallStack)
+import Text.Show.Pretty (ppShow)
+
+import qualified Clash.Netlist.Id as Id
+import qualified Clash.Netlist.Types as N
+import qualified Clash.Primitives.DSL as DSL
+
+xpmCdcArraySingleTF :: TemplateFunction
+xpmCdcArraySingleTF =
+  TemplateFunction
+    [registerInput, initBehavior, stages, clkSrc, clkDst, input]
+    (const True)
+    xpmCdcArraySingleTF#
+ where
+  _2LteN
```

Comment:
```suggestion
  _2LteStages
```

I didn't say anything about it in `xpmCdcSingle`, but given that you now also have `1LteN` and `nLte1024`, it gets confusing :-)